chore(deps): upgraded prod dependencies

nikomakela opened the pull request:
HH-372 TH-1376 LIIKUNTA-704.

Upgraded many of the production dependencies to the latest.

Changed the escheck to check that all code is es2022 equivalent, which
makes the es-check to pass. The check was earlier for es2020.

Reinitialized Husky since it was giving a deprecation note.

Set resolution for react 18: Fix for
"venue-graphql-proxy@workspace:proxies/venue-graphql-proxy provides
react (pcf21b) with version 18.3.1, which doesn't satisfy what
@events-helsinki/eslint-config-bases and some of its descendants
request".

Set proper peer dependency for eslint: The peer dependency did not match
with the actually installed eslint.

Fix some yarn warnings: Fixed multiple dependency warnings where the
provided package did not properly satisfy the requests of other
dependencies.

Fixed Browser tests: Browser tests needs different translations for
different kind of searches.

Fixed (esm) build issues in Venue-graphql-proxy.

[skip lint-staged task if there are no files associated for it](https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/787/commits/a600a3ffa85af75dc31396fd02b640db4ebb95af): For example the `yarn prettier --write` fails if there are no filepaths given for it. Because of that, the Husky pre-commit fails. This commit skips the task if it cannot find any staged files for the task.

nikomakela committed:
chore(deps): upgraded dependencies

HH-372 TH-1376 LIIKUNTA-704.

Upgraded many of the production dependencies to the latest.

Changed the escheck to check that all code is es2022 equivalent, which
makes the es-check to pass. The check was earlier for es2020.

Reinitialized Husky since it was giving a deprecation note.

Set resolution for react 18: Fix for
"venue-graphql-proxy@workspace:proxies/venue-graphql-proxy provides
react (pcf21b) with version 18.3.1, which doesn't satisfy what
@events-helsinki/eslint-config-bases and some of its descendants
request".

Set proper peer dependency for eslint: The peer dependency did not match
with the actually installed eslint.

Fix some yarn warnings: Fixed multiple dependency warnings where the
provided package did not properly satisfy the requests of other
dependencies.

Fixed Browser tests: Browser tests needs different translations for
different kind of searches.

Fixed (esm) build issues in Venue-graphql-proxy.

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr787.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr787.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr787.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
chore: skip lint-staged task if there are no files associated for it

HH-372 TH-1376 LIIKUNTA-704.

For example the `yarn prettier --write` fails if there are no filepaths
given for it. Because of that, the Husky pre-commit fails. This commit
skips the task if it cannot find any staged files for the task.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr787.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr787.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr787.dev.hel.ninja :laughing::tada::tada::tada:
